libsolidity/interface/CompilerStack.cpp

```diff
@@ -1246,7 +1257,7 @@ void CompilerStack::generateIR(ContractDefinition const& _contract)
 	for (auto const& pair: m_contracts)
 		otherYulSources.emplace(pair.second.contract, pair.second.yulIR);
 
-	IRGenerator generator(m_evmVersion, m_revertStrings, m_optimiserSettings);
+	IRGenerator generator(m_evmVersion, m_revertStrings, m_optimiserSettings, m_contractCallGraphs[&_contract]);
```

Comment:
It looks weird to drag this through just to verify it later on. Would it work to call something like`generator.verifyCallGraph(m_contractCallGraphs[&_contract]))` after `generator.run(...)`?

Reply:
I would need to save the lists of generated functions in that case

Reply:
I mean, in the generator I need to remember what functions where generated so I can compare. Currently that list is gone the moment we call `requestedFunctions()`

Reply:
I'm talking about the Solidity functions - IRGenerator::m_functionGenerationQueue

Reply:
That's `IRGeneratorContext`, but my point stands, that list is empty when the `run` function returns because all functions have been generated at that point so the queue has nothing to process anymore..

Reply:
Sure, you have to store for which ones we generated code.